libraries/AP_Math/control.cpp

```diff
@@ -407,3 +407,16 @@ float kinematic_limit(Vector3f direction, float max_xy, float max_z_pos, float m
     }
     return fabsf(max_z_neg/direction.z);
 }
+
+// calculate the expo function on the normalised input
+// input must be in the range of -1 to 1
+// expo should be less than 1.0 but limited to be less than 0.95
+float input_expo(float input, float expo)
+{
+    input = constrain_float(input, -1.0, 1.0);
+    if (expo < 0.95) {
+        return (1 - expo) * input / (1 - expo * fabsf(input));
+    } else {
```

Comment:
nitpick: because we return early if expo < 0.95 we don't need the else {}.  we can just return input; after the if